src/Riok.Mapperly/Descriptors/DescriptorBuilder.cs

```diff
@@ -63,8 +64,8 @@ MapperConfiguration defaultMapperConfiguration
     {
         ConfigureMemberVisibility();
         ReserveMethodNames();
-        ExtractObjectFactories();
         ExtractUserMappings();
+        ExtractObjectFactories();
```

Comment:
As the order has changed here, the documentation in `architecture.md` needs to be adjusted too.